src/mpi/init/init.c

```diff
@@ -179,6 +179,9 @@ int MPI_Init( int *argc, char ***argv )
     if (mpi_errno != MPI_SUCCESS) goto fn_fail;
 
     if (MPIR_CVAR_ASYNC_PROGRESS) {
+#if MPL_THREAD_PACKAGE_NAME == MPL_THREAD_PACKAGE_ARGOBOTS
+        MPL_error_printf("Asynchronous progress is not supported with Argobots\n");
```

Comment:
This message is more of a warning than an error. I think we should abort the execution.